engine/skiplist.cpp

```diff
@@ -136,9 +71,30 @@ Status Skiplist::Rebuild() {
   return Status::Ok;
 }
 
-void Skiplist::Seek(const StringView &key, Splice *result_splice) {
+void Skiplist::Seek(const StringView &key, StringView value,
+                    Splice *result_splice) {
   result_splice->seeking_list = this;
-  header_->SeekNode(key, header_->Height(), 1, result_splice);
+  if (!cmp_ctx.priority_key && value.empty()) {
```

Comment:
Don't search hash table in seek. 
For sset operation, if key exist, it won't call Seek(), this will add extra overhead for insert operation.

Reply:
Though this maybe useful for iterator seek, maybe split to another function.

Reply:
I don't want to add user API for compare now.  if we want to increase ranged search by value, we can split two function, and add user API. But now,  it is no need. 
By the way,  this code only used when sorted by value and meanwhile seek by key.  So for `sset` operation, if key exist, this can't add extra overhead for insert operation.

Reply:
I still can't understand why we need seek by key while nodes are sorted by values

Reply:
BTW, you need to modify sorted iterator to support range scan by value

Reply:
I know,  pls you see  here `seek` as `seekbykey`

Reply:
I mean, how to seek(by key or by value) should be determined by what comparator the skiplist is using, if skiplist is sorted by value, then don't seek by key ever, vise versa.

Reply:
{student, score},  sorted by score, why can't seek by student?

Reply:
Seek is used to iterate data ordered, as data is sorted by score, we can't iterate student ordered, and it's meanless too.

If need get a certain student, just call sget.